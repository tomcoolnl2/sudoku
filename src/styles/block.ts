
import  styled, { css } from 'styled-components'


interface BlockContainerProps {
    selected?: boolean
    highlighted?: boolean
    clue?: boolean
}

export const BlockContainer = styled.div<BlockContainerProps>`
    ${({ theme, clue, selected, highlighted }) => css`
        align-items: center;
        background-color: ${theme.colors[selected ? 'active' : (highlighted ? 'highlighted' : 'white')]};
        border: 1px solid ${theme.colors.black};
        cursor: pointer;
        display: flex;
        flex-grow: 1;
        flex-shrink: 0;
        flex-basis: 0;
        font-size: 20px;
        font-weight: ${clue ? 'bold' : 'normal'};
        height: auto;
        justify-content: center;
        transition: ${theme.transition};
        user-select: none;

        &::before {
            padding-top: 100%;
            content: '';
            float: left;
        }

        &:hover {
            background-color: ${theme.colors.highlighted}
        }
    `}
`