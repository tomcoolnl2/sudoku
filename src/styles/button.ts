
import  styled, { css } from 'styled-components'


export const Button = styled.button`
    ${({ theme }) => css`
        align-items: center;
        background-color: ${theme.colors.secondary};
        border: 2px solid ${theme.colors.secondary};
        border-radius: 4px;
        color: ${theme.colors.primary};
        cursor: pointer;
        display: flex;
        flex: 1;
        font-size: 16px;
        height: 40px;
        justify-content: center;
        margin: 4px 2px;
        min-height: 40px;
        padding: 0;
        transition: ${theme.transition};
        opacity: .9;

        &:disabled {
            opacity: .7;
            cursor: default;
        }

        &:focus {
            border-color: ${theme.colors.active};
            outline: none;
        }

        &:hover {
            opacity: .7;
        }

        > svg { 
            margin-right: 10px;
            fill: ${theme.colors.primary};
            position: relative;
            width: 24px;
            height: 24px;
        }
    `}
`