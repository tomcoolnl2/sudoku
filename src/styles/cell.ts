
import  styled, { css } from 'styled-components'


interface CellContainerProps {
    selected?: boolean
    highlighted?: boolean
    clue?: boolean
    duplicate?: boolean
    mistake?: boolean
}

export const CellContainer = styled.div<CellContainerProps>`
    ${({ theme, clue, selected, highlighted, duplicate, mistake }) => css`
        align-items: center;
        background-color: ${theme.colors[selected || duplicate ? 'active' : (highlighted ? 'highlighted' : 'primary')]};
        border: 1px solid ${theme.colors.secondary};
        cursor: pointer;
        color: ${mistake ? 'red' : 'inherit'};
        display: flex;
        flex-grow: 1;
        flex-shrink: 0;
        flex-basis: 0;
        font-size: 20px;
        font-weight: ${clue ? 'bold' : 'normal'};
        height: auto;
        justify-content: center;
        transition: ${theme.transition};
        user-select: none;

        &::before {
            padding-top: 100%;
            content: '';
            float: left;
        }

        &:hover {
            background-color: ${theme.colors[selected ? 'active' : 'highlighted']}
        }
    `}
`