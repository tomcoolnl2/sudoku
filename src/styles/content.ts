
import styled from 'styled-components'


export const Content = styled.div`
    max-width: 500px;
    flex: 1;
    flex-direction: column;
    justify-content: center;
`