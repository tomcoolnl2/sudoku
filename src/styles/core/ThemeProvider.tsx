
import { FC } from 'react'
import { useSelector } from 'react-redux'
import { ThemeProvider } from 'styled-components'
import { lightTheme, darkTheme, GlobalStyles } from './'
import { StoreReducer } from '../../redux'


export interface DarkThemeProviderState {
    darkModeEnabled: boolean
}

export const DarkThemeProvider: FC = ({ children }) => {
	const { darkModeEnabled } = useSelector<StoreReducer, DarkThemeProviderState>(state => state.settings)
	const theme = darkModeEnabled ? darkTheme : lightTheme
	return (
		<ThemeProvider theme={theme}>
			<GlobalStyles />
			{children}
		</ThemeProvider>
	)
}