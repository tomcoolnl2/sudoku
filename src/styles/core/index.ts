
import { createGlobalStyle, css } from 'styled-components'


export const theme = {
	colors: {
		background: 'radial-gradient(#282c34cc, #282c34)',
		black: '#282c34',
		blue: '#a0e9fd',
		lightBlue: '#caf3fe',
		white: 'white'
	},
	transition: '0.3s'
}

export const GlobalStyles = createGlobalStyle`
    ${props => css`
        html {
            height: 100%;
        }

        body {
            display: flex;
            flex-direction: column;
            height: 100%;
            margin: 0;

            #root {
                background-image: ${props.theme.colors.background};
                color: ${props.theme.colors.black};
                display: flex;
                font-family: sans-serif;
                height: 100%;
                justify-content: center;
                padding: 15px;
            }
        }
    `
}`