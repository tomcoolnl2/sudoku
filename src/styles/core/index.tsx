
import { createGlobalStyle, css } from 'styled-components'

const theme = {
	colors: {
		background: 'radial-gradient(#282c34cc, #282c34)',
		primary: 'white',
		secondary: '#282c34',
		active: '#EDDA74',
		highlighted: '#FCF3CF',
		errorPrimary: '#CB4335',
		errorSecundary: '#E6B0AA'
	},
	transition: '0.3s'
}

export const lightTheme = {
	...theme
}

export const darkTheme = {
	...theme,
	colors: {
		...theme.colors,
		primary: '#2E4053',
		secondary: '#EBEDEF',
		active: '#515A5A',
		highlighted: '#616A6B',
	},
}

export const GlobalStyles = createGlobalStyle`
    ${props => css`
		:root {
			color-scheme: light dark;
		}

        html {
            height: 100%;
        }

        body {
            display: flex;
            flex-direction: column;
            height: 100%;
            margin: 0;

            #root {
                background-image: ${props.theme.colors.background};
                color: ${props.theme.colors.secondary};
                display: flex;
                font-family: sans-serif;
                height: 100%;
                justify-content: center;
                padding: 15px;
            }
        }
    `
}`