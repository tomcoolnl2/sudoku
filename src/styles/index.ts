
export * from './title'
export * from './content'
export * from './title'
export * from './card'
export * from './grid'
export * from './block'