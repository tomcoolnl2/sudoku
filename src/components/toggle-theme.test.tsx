
import { render, screen } from '@testing-library/react'
import { withTheme } from '../styles/core'
import { ToggleTheme, ToggleThemeProps } from './'


const props: ToggleThemeProps = {
	selected: false,
	toggleTheme: jest.fn()
}

describe('ToggleTheme', () => {
	
	it('renders with light mode as default', () => {
		const { container } = render(withTheme(<ToggleTheme {...props} />))
		expect(container.firstChild).toMatchSnapshot()
	})

	it('should be present within the DOM', () => {
		render(withTheme(<ToggleTheme {...props} />))
		const buttonElement = screen.getByText(/Dark mode/i)
		expect(buttonElement).toBeInTheDocument()
	})
})