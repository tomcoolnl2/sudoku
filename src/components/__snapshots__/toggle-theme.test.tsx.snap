// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleTheme should match it's snapshot 1`] = `
<label
  for="theme-toggle"
>
  Dark mode: 
  <input
    name="theme-toggle"
    type="checkbox"
  />
</label>
`;
